flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/SourceOperator.java

```diff
@@ -197,7 +195,20 @@ public void sendSourceEventToCoordinator(SourceEvent event) {
 
 			@Override
 			public UserCodeClassLoader getUserCodeClassLoader() {
-				return (UserCodeClassLoader) getRuntimeContext().getUserCodeClassLoader();
+				return new UserCodeClassLoader() {
```

Comment:
Is it possible to reuse code from `RuntimeContextInitializationContextAdapters` here?

Reply:
That was my first attempt as well. But it seems a little ugly for two reasons:
1. The `RuntimeContextInitializationContextAdapters` is itself a class converting `RuntimeContext` to `InitializationContext`. The scope and usage of that class is quite clear. The `RuntimeContextUserCodeClassLoaderAdapter` is a nested private class. Making only that class public seems a little weird.
2. The `RuntimeContextInitializationContextAdapters` is affiliated with `DeserializationSchema` which is something that the SourceOperator should not be aware of.

Another 2 options are:
1. Move the `RuntimeContextUserCodeClassLoaderAdapter` to a separate class.
2. Add a new method to `RuntimeContext` which returns an actual `UserCodeClassLoader` instead of letting the caller construct by themselves.

Both approach seems a little unnecessary here because the inline class is quite simple.